src/util/triplea/MapXMLCreator/MapPropertiesPanel.java

```diff
@@ -0,0 +1,360 @@
+import javax.swing.JTextField;
+
+import util.image.FileOpen;
+
+/**
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+public class MapPropertiesPanel {
+	
+	public static void layout(final MapXMLCreator mapXMLCreator, final JPanel m_stepActionPanel) {
+
+		if (!MapXMLHelper.s_xmlStrings.containsKey("info_@name"))
+		{
+			for (final String newKey : new String[] { "info_@name", "info_@version" })
+			{
+				MapXMLHelper.s_xmlStrings.put(newKey, "");
+			}
+		}
+		
+		if (MapXMLHelper.s_resourceList.isEmpty())
+			MapXMLHelper.s_resourceList.add("");
+
+		final JTextField tMapName = new JTextField(MapXMLHelper.s_xmlStrings.get("info_@name"));
+		final JTextField tMapVersion = new JTextField(MapXMLHelper.s_xmlStrings.get("info_@version"));
+		final JTextField tResourceName = new JTextField(MapXMLHelper.s_resourceList.get(0));
+		final JTextField tMapImageFile = new JTextField((MapXMLCreator.s_mapImageFile==null?"":MapXMLCreator.s_mapImageFile.getAbsolutePath()));
+		final JTextField tCentersFile = new JTextField((MapXMLCreator.s_mapCentersFile==null?"":MapXMLCreator.s_mapCentersFile.getAbsolutePath()));
+		final JTextField tWaterFilter = new JTextField(MapXMLCreator.s_waterFilterString);
+		final JLabel lMapName = new JLabel("Map Name:");
+		final JLabel lMapNameExample = new JLabel("e.g. 'Revised'");
+		final JLabel lMapVersion = new JLabel("Map Version:");
+		final JLabel lMapVersionExample = new JLabel("e.g. '1.2.0.1'");
+		final JLabel lResourceName = new JLabel("Resource Name:");
+		final JLabel lResourceNameExample = new JLabel("e.g. 'PUs'");
+		final JLabel lMapImageFile = new JLabel("Map Image File:");
+		final JButton bSelectMapImageFile = new JButton("Browse");
+		final JLabel lCentersFile = new JLabel("Map Centers File:");
+		final JButton bSelectCentersFile = new JButton("Browse");
+		final JLabel lWaterFilter = new JLabel("Water Territory Filter");
+		final JLabel lWaterFilterExample = new JLabel("e.g. 'Sea Zone'");		
+		
+		GridBagLayout gbl_panel_1 = new GridBagLayout();
+		gbl_panel_1.rowHeights = new int[] { 32, 32, 32, 32, 32, 32, 32, 0 };
+		gbl_panel_1.rowWeights = new double[] { 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, Double.MIN_VALUE };
+		m_stepActionPanel.setLayout(gbl_panel_1);
+		
+		//Map Name
+		GridBagConstraints gbc_lMapName = new GridBagConstraints();
+		gbc_lMapName.insets = new Insets(0, 0, 5, 5);
+		gbc_lMapName.anchor = GridBagConstraints.NORTHEAST;
+		gbc_lMapName.gridheight = 32;
+		gbc_lMapName.gridy = 1;
+		gbc_lMapName.gridx = 0;
+		m_stepActionPanel.add(lMapName, gbc_lMapName);
+
+		tMapName.addFocusListener(new FocusListener() {
+			
+			@Override
+			public void focusLost(FocusEvent e) {
+				MapXMLHelper.s_xmlStrings.put("info_@name", tMapName.getText());
+			}
+			
+			@Override
+			public void focusGained(FocusEvent e) {
+			}
+		});
+		tMapName.setMaximumSize(new Dimension(0, 0));
+		tMapName.setColumns(30);
+		GridBagConstraints gbc_m_tMapName = (GridBagConstraints) gbc_lMapName.clone();

```

Comment:
Hey, I really appreciate you putting so much effort into "giving me a hard time"! :)
At the end it should benefit the project and that what it is about.
I will remove the 'm_' which was coming from a time the gbc was for a member variable m_tMapName. However, I would like to keep the gbc_ prefix as it is an easy to spot prefix and also helpful to see what the gbc object is build for.